Make September 2025 news public, add October template (#31608)
idaholab/moose: modules/doc/content/newsletter/2025/2025_09.md

[]
```

> In this case, the test `needs_cuda` will be run beacuse it depends on the `cuda` capability, while

Reviewer: ```suggestion
In this case, the test `needs_cuda` will be run because it depends on the `cuda` capability, while
```

the `doesnt_need_cuda` test will not run because it does not depend on the `cuda` capability.

### Optimizations

> Face material properties are now only computed if the boundary conditions, interface kernels, and/or
> constraints active on the relevant boundary have requested material properties. The logic for deciding

Reviewer: ```suggestion
Face and neighbor face material properties are now only computed if the boundary conditions, interface kernels active on the
relevant boundary have requested material properties, or if any active constraint has requested a material property. The logic for deciding
```

- Added optional parameter to `build_side_list_from_node_list()` to allow only some nodesets to be
  converted rather than all
- Added `Parameters::Value::type_info()` method
- Updated autotools to their latest versions.  This fixes a linker bug that was preventing libMesh
  compilation with some newer software stacks.
- Use `MatGetValue` instead of `MatGetRow` in `PetscMatrix::operator()`, to improve compatibility and
  efficiency
- `PerfLog` performance logging of Netgen tetrahedralization
- Bug fixes for parallel `StaticCondensation` operation
- Bug fix for certain use cases of `PetscVector` constructed directly from PETSc `Vec`
- Bug fix for triangulation of boundaries with quadratic edges
- Bug fix for an adaptive refinement failure case on distributed meshes
- Fixed handling of linearly-dependent data in reduced basis EIM training
- Fixed small `PetscViewer` memory leaks in `print_matlab()` methods

## PETSc-level Changes

## Bug Fixes, Minor Changes, and Minor Enhancements

- [PythonFMU](https://github.com/NTNU-IHB/PythonFMU) has been added to the `moose-dev` container using
  GCC.
- Added an Intel LLVM compiler container option.

> - A [GenericConstantStdVectorMaterial.md] was added to use `std::vector<Real>` properties

Reviewer: ```suggestion
- A [GenericConstantStdVectorMaterial.md] was added to use `std::vector<Real>` properties.
  It is modeled after the [GenericConstantVectorMaterial.md]. The `GenericConstantRealVectorValue` has been deprecated.
```
